hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/processors/StreamToStreamJoinP.java

```diff
@@ -102,12 +104,14 @@ public class StreamToStreamJoinP extends AbstractProcessor {
 
     @SuppressWarnings("checkstyle:ExecutableStatementCount")
     public StreamToStreamJoinP(
+            final int processorIndex,
             final JetJoinInfo joinInfo,
             final Map<Byte, ToLongFunctionEx<JetSqlRow>> leftTimeExtractors,
             final Map<Byte, ToLongFunctionEx<JetSqlRow>> rightTimeExtractors,
             final Map<Byte, Map<Byte, Long>> postponeTimeMap,
             final Tuple2<Integer, Integer> columnCounts
     ) {
+        this.processorIndex = processorIndex;
```

Comment:
We don't need to pass the processor index as an argument. It's available in the `context` in the `init()` method below.